server/py/framework/db/sqldb/db.py

```diff
@@ -4691,6 +4691,18 @@ def _upsert(self, session, objects, ignore=False, silent=False):
             session.add(object_)
         self._commit(session, objects, ignore, silent)
 
+    def _upsert_object_and_flush_to_get_field(self, session, object_, field):
+        # Add the object to the session
+        session.add(object_)
+        # Flush the session to generate the database values
+        session.flush()
```

Comment:
Why do you need to explicitly use `flush()` here? Since you're committing to DB anyway later in this function, wouldn't doing the `getattr` after the `commit` behave exactly the same?

Reply:
@theSaarco yeah, that's what I expected, but after checking it - it didn't work for some reason(`id` was empty after committing )